### PR TITLE
pg:fix potential bug, in recover process, should dequeue rw lock waiters before other queue objs

### PR DESCRIPTION
**Pull request by yanqiang-ux**
fix potential bug, in func on_global_recover should release obj rwlock at last, because the obj will block in rwlock waiters firstly with the smaller tid, and should dequeue rwlock waiters before other queue.

Signed-off-by: yanqiang <yanqiang@xsky.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by yanqiang-ux** [comment]
@neha-ojha @tchaikov
pls review this commit, thx.

**Comment by LiumxNL** [comment]
unittest_rbd_mirror failed, not related to this pr.

**Comment by LiumxNL** [comment]
test this please

**Comment by neha-ojha** [comment]
jenkins retest this please

File: src/osd/PrimaryLogPG.cc [hunk 1]
```diff
@@ -506,6 +493,19 @@ void PrimaryLogPG::on_global_recover(
     waiting_for_unreadable_object.erase(unreadable_object_entry);
   }
   finish_degraded_object(soid);
+  // release obj rwlock after release degraded and unreadable_object, in some case will casuse out-of-order issue.
```
Comment (neha-ojha) [inline src/osd/PrimaryLogPG.cc:496]: Have you encountered an out-of-order bug due to this? If so, it will be good to add tracker issue with details.

**Comment by tchaikov** [comment]
- https://pulpito.ceph.com/kchai-2020-08-25_04:54:49-rados-wip-kefu-testing-2020-08-25-0950-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2020-08-26_16:53:29-rados-wip-kefu-testing-2020-08-26-1944-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2020-08-27_03:37:15-rados-wip-kefu-testing-2020-08-27-0139-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2020-08-27_08:22:12-rados-wip-kefu-testing-2020-08-27-1255-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2020-08-27_10:01:54-rados-wip-kefu-testing-2020-08-27-1255-distro-basic-smithi/

**Comment by tchaikov** [comment]
@yanqiang-ux ping?

File: src/osd/PrimaryLogPG.cc (hunk 1 above)
Comment (jdurgin) [reply src/osd/PrimaryLogPG.cc:496]: any more details on how this was found? did it cause an issue in a running system, or was it by code inspection?

**Comment by ljflores** [comment]
@yanqiang-ux any updates on this PR?

**Pull request closed without merging**